### ISSUE TITLE
Add parser support for SQLite's undocumented inline AUTOINCREMENT syntax

### ISSUE THREAD

**Issue by KweezyCode** [jOOQ/jOOQ#16216]
### Expected behavior

working code generator

### Actual behavior

```
15:55:09 SEVERE DDLDatabase Error        : Your SQL string could not be parsed or interpreted. This may have a variety of reasons, including:
- The jOOQ parser doesn't understand your SQL
- The jOOQ DDL simulation logic (translating to H2) cannot simulate your SQL

If you think this is a bug or a feature worth requesting, please report it here: https://jooq.org/bug

As a workaround, you can use the Settings.parseIgnoreComments syntax documented here:
https://www.jooq.org/doc/latest/manual/sql-building/dsl-context/custom-settings/settings-parser/
15:55:09 SEVERE Error while loading file: ./src/main/resources/ddl.sql
15:55:09 SEVERE An exception occurred while parsing script source : ./src/main/resources/ddl.sql. Please report this error to https://jooq.org/bug
org.jooq.impl.ParserException: Token ')' expected: [15:19] ...hecked INT NOT NULL DEFAULT 0,
	PRIMARY KEY("id" [*]AUTOINCREMENT)
);

-- INDEX
 
-- TRIGGER
 
-- VIEW
 

	at org.jooq.impl.DefaultParseContext.expected(ParserImpl.java:15017)
	at org.jooq.impl.DefaultParseContext.parse(ParserImpl.java:14224)
	at org.jooq.impl.DefaultParseContext.parse(ParserImpl.java:14219)
	at org.jooq.impl.DefaultParseContext.parseParenthesisedSortSpecification(ParserImpl.java:6490)
	at org.jooq.impl.DefaultParseContext.parseKeyColumnList(ParserImpl.java:5291)
	at org.jooq.impl.DefaultParseContext.parsePrimaryKeySpecification(ParserImpl.java:5261)
	at org.jooq.impl.DefaultParseContext.parseCreateTable(ParserImpl.java:4572)
	at org.jooq.impl.DefaultParseContext.parseCreate(ParserImpl.java:2962)
	at org.jooq.impl.DefaultParseContext.parseQuery(ParserImpl.java:1136)
	at org.jooq.impl.DefaultParseContext.lambda$parse$0(ParserImpl.java:906)
	at org.jooq.impl.DefaultParseContext.wrap(ParserImpl.java:15244)
	at org.jooq.impl.DefaultParseContext.parse(ParserImpl.java:894)
	at org.jooq.impl.ParserImpl.parse(ParserImpl.java:772)
	at org.jooq.impl.ParserImpl.parse(ParserImpl.java:767)
	at org.jooq.meta.extensions.ddl.DDLDatabase.load(DDLDatabase.java:179)
	at org.jooq.meta.extensions.ddl.DDLDatabase.lambda$export$0(DDLDatabase.java:166)
	at org.jooq.FilePattern.load0(FilePattern.java:310)
	at org.jooq.FilePattern.load(FilePattern.java:290)
	at org.jooq.FilePattern.load(FilePattern.java:242)
	at org.jooq.meta.extensions.ddl.DDLDatabase.export(DDLDatabase.java:166)
	at org.jooq.meta.extensions.AbstractInterpretingDatabase.connection(AbstractInterpretingDatabase.java:100)
	at org.jooq.meta.extensions.AbstractInterpretingDatabase.create0(AbstractInterpretingDatabase.java:77)
	at org.jooq.meta.AbstractDatabase.create(AbstractDatabase.java:408)
	at org.jooq.meta.AbstractDatabase.create(AbstractDatabase.java:398)
	at org.jooq.meta.AbstractDatabase.setConnection(AbstractDatabase.java:376)
	at org.jooq.codegen.GenerationTool.run0(GenerationTool.java:566)
	at org.jooq.codegen.GenerationTool.run(GenerationTool.java:246)
	at org.jooq.codegen.GenerationTool.generate(GenerationTool.java:241)
	at org.jooq.codegen.GenerationTool.main(GenerationTool.java:213)

15:55:09 SEVERE Error in file: /home/kweezy/IdeaProjects/Rasphost/sources/Lobby/IceShield/build/tmp/generateJooq/config.xml. Error : Error while exporting schema
org.jooq.exception.DataAccessException: Error while exporting schema
	at org.jooq.meta.extensions.AbstractInterpretingDatabase.connection(AbstractInterpretingDatabase.java:103)
	at org.jooq.meta.extensions.AbstractInterpretingDatabase.create0(AbstractInterpretingDatabase.java:77)
	at org.jooq.meta.AbstractDatabase.create(AbstractDatabase.java:408)
	at org.jooq.meta.AbstractDatabase.create(AbstractDatabase.java:398)
	at org.jooq.meta.AbstractDatabase.setConnection(AbstractDatabase.java:376)
	at org.jooq.codegen.GenerationTool.run0(GenerationTool.java:566)
	at org.jooq.codegen.GenerationTool.run(GenerationTool.java:246)
	at org.jooq.codegen.GenerationTool.generate(GenerationTool.java:241)
	at org.jooq.codegen.GenerationTool.main(GenerationTool.java:213)
Caused by: org.jooq.impl.ParserException: Token ')' expected: [15:19] ...hecked INT NOT NULL DEFAULT 0,
	PRIMARY KEY("id" [*]AUTOINCREMENT)
);

-- INDEX
 
-- TRIGGER
 
-- VIEW
 

	at org.jooq.impl.DefaultParseContext.expected(ParserImpl.java:15017)
	at org.jooq.impl.DefaultParseContext.parse(ParserImpl.java:14224)
	at org.jooq.impl.DefaultParseContext.parse(ParserImpl.java:14219)
	at org.jooq.impl.DefaultParseContext.parseParenthesisedSortSpecification(ParserImpl.java:6490)
	at org.jooq.impl.DefaultParseContext.parseKeyColumnList(ParserImpl.java:5291)
	at org.jooq.impl.DefaultParseContext.parsePrimaryKeySpecification(ParserImpl.java:5261)
	at org.jooq.impl.DefaultParseContext.parseCreateTable(ParserImpl.java:4572)
	at org.jooq.impl.DefaultParseContext.parseCreate(ParserImpl.java:2962)
	at org.jooq.impl.DefaultParseContext.parseQuery(ParserImpl.java:1136)
	at org.jooq.impl.DefaultParseContext.lambda$parse$0(ParserImpl.java:906)
	at org.jooq.impl.DefaultParseContext.wrap(ParserImpl.java:15244)
	at org.jooq.impl.DefaultParseContext.parse(ParserImpl.java:894)
	at org.jooq.impl.ParserImpl.parse(ParserImpl.java:772)
	at org.jooq.impl.ParserImpl.parse(ParserImpl.java:767)
	at org.jooq.meta.extensions.ddl.DDLDatabase.load(DDLDatabase.java:179)
	at org.jooq.meta.extensions.ddl.DDLDatabase.lambda$export$0(DDLDatabase.java:166)
	at org.jooq.FilePattern.load0(FilePattern.java:310)
	at org.jooq.FilePattern.load(FilePattern.java:290)
	at org.jooq.FilePattern.load(FilePattern.java:242)
	at org.jooq.meta.extensions.ddl.DDLDatabase.export(DDLDatabase.java:166)
	at org.jooq.meta.extensions.AbstractInterpretingDatabase.connection(AbstractInterpretingDatabase.java:100)
	... 8 more


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':IceShield:generateJooq'.
> Process 'command '/usr/lib/jvm/java-17-openjdk-amd64/bin/java'' finished with non-zero exit value 255

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 2s
```

### Steps to reproduce the problem

```sql
CREATE TABLE "authme" (
	"id"	INTEGER,
	"username"	VARCHAR(255) NOT NULL UNIQUE,
	"realname"	VARCHAR(255) NOT NULL DEFAULT 'Player',
	"password"	VARCHAR(255) NOT NULL,
	"ip"	VARCHAR(40) NOT NULL,
	"firstlogin"	BIGINT NOT NULL DEFAULT 0,
	"lastlogin"	BIGINT NOT NULL DEFAULT 0,
	"x"	DOUBLE NOT NULL DEFAULT '0.0',
	"y"	DOUBLE NOT NULL DEFAULT '0.0',
	"z"	DOUBLE NOT NULL DEFAULT '0.0',
	"world"	VARCHAR(255) NOT NULL DEFAULT 'world',
	"isLogged"	INT DEFAULT '0', verified INT NOT NULL DEFAULT 0, newsChecked INT NOT NULL DEFAULT 0,
	PRIMARY KEY("id" AUTOINCREMENT)
);
```

AUTOINCREMENT key causes problem, without it jooq generator works fine.

Config:
```
jooq {
    version.set("3.19.3")  // default (can be omitted)
    edition.set(nu.studer.gradle.jooq.JooqEdition.OSS)  // default (can be omitted)

    configurations {
        create("main") {  // name of the jOOQ configuration
            generateSchemaSourceOnCompilation.set(true)  // default (can be omitted)

            jooqConfiguration.apply {
                logging = Logging.WARN

                generator.apply {
                    name = "org.jooq.codegen.DefaultGenerator"
                    database.apply {
                        name = "org.jooq.meta.extensions.ddl.DDLDatabase"
                        properties.add(Property().apply {
                            key = "scripts"
                            value = "./src/main/resources/ddl.sql"
                        })

                    }
                    generate.apply {
                        isDeprecated = false
                        isRecords = false
                        isImmutablePojos = false
                        isFluentSetters = false
                    }
                    target.apply {
                        packageName = "nu.studer.sample"
                        directory = "./build/generated-src/jooq/main"  // default (can be omitted)
                    }
                    strategy.name = "org.jooq.codegen.DefaultGeneratorStrategy"
                }
            }
        }
    }
}
```

### jOOQ Version

OSS 3.19.3

### Database product and version

SQL struct (of SQLite DB)

### Java Version

openjdk version "17.0.9" 2023-10-17

### OS Version

Kubuntu 23.10

### JDBC driver name and version (include name if unofficial driver)

missing

**Comment by KweezyCode** [comment]
should i close ticket?

**Comment by lukaseder** [comment]
No, I'll support the syntax in our parser. It shouldn't be too hard.

**Comment by lukaseder** [comment]
Implemented in version 3.20.0
